core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -595,19 +596,22 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     // Precompute this because mPos will be updated several times in this function.
     final boolean isInCurrentBlock = pos / mBlockSize == mPos / mBlockSize;
 
-    // No need to cache if the stream hasn't read anything and the seek position is not within the
-    // block
-    if (mPos != 0 || mCurrentBlockInStream != null || isInCurrentBlock) {
+    // No need to cache if (1) the stream hasn't read anything and the seek position is not within
+    // the block, or (2) the in stream reads from the local worker
+    boolean firstReadAcrossBlock = mPos == 0 && mCurrentBlockInStream == null && !isInCurrentBlock;
+    boolean readFromLocalWorker = mCurrentBlockInStream != null
+        && mCurrentBlockInStream.Source() == BlockInStreamSource.LOCAL;
+    if (!firstReadAcrossBlock && !readFromLocalWorker) {
```

Comment:
was the missing local worker part a bug?

Reply:
No, this condition was in `readCurrentBlockToPos ` and now moved outside.